src/frontends/lean/tactic_notation.cpp

```diff
@@ -141,7 +142,12 @@ static expr parse_interactive_param(parser & p, expr const & ty, expr const & qu
     auto env = eval.compile(n, quote_inst);
     vm_state S(env, p.get_options());
     auto vm_res = S.invoke(n, vm_parsed);
-    return to_expr(vm_res);
+    expr r = to_expr(vm_res);
+    if (is_app_of(r, get_pexpr_subst_name())) {
+        return r; // HACK
```

Comment:
This hack is due to the fact that we cannot actually reflect all `pexpr`s - to do that we would have to know which `to_pexpr` instance to use for each antiquotation. I propose to have `pexpr.reflect` only work with `pexpr` antiquotations, which in practice just means that you cannot use the `[...] tactic quotations with `expr` antiquotations.